javaslang/generator/Generator.scala

```diff
@@ -1452,11 +1529,35 @@ def generateTestClasses(): Unit = {
       val AssertionsExtensions = im.getType("javaslang.AssertionsExtensions")
       val ListType = im.getType("javaslang.collection.List")
       val OptionType = im.getType("javaslang.control.Option")
+      val FutureType = im.getType("javaslang.concurrent.Future")
+      val ExecutorServiceType = im.getType("java.util.concurrent.Executors")
+      val ExecutorService = ExecutorServiceType + ".newSingleThreadExecutor()"
 
       val d = "$";
 
       im.getStatic("javaslang.API.*")
 
+      def genFutureTests(name: String, value: String, success: Boolean): String = {
+        val check = if (success) "isSuccess" else "isFailure"
+        xs"""
+          @$test
+          public void shouldFutureWith${name}ReturnNotNull() {
+              $FutureType<Object> future = Future($value);
+              future.await();
```

Comment:
@danieldietrich, if `await` would return the parent `Future`, we could eliminate the need for this line, i.e. 
```java
$FutureType<Object> future = Future($value).await();
```